ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js

```diff
@@ -57,6 +57,21 @@
                     FormView.customFunctions.defaultSubmitErrorHandler(defaultSubmissionError, globals);
                 }
             }, "submitError");
+            this._model.subscribe((action) => {
+                let state = action.target.getState();
+                // execute the handler only if there are no rules configured on submitSuccess event.
+                if (!state.events['custom:saveSuccess'] || state.events['custom:saveSuccess'].length === 0) {
```

Comment:
This looks incorrect to me, you have hardcoded the view here and the same code would not work for headless too. Can you change the code and implement it the same way as done in submit ? Additionallly, you also need to fix this, https://github.com/adobe/react-starter-kit-aem-headless-forms/blob/main/src/Form.tsx#L50-L83

Reply:
As of now, we are hardcoding save success and failure event and not asking from customer. 
And, here we are checking if these events are registered through rule (save success and save error), then we are not executing default functionality. Submit is also implemented the same way here.

https://github.com/adobe/react-starter-kit-aem-headless-forms/blob/main/src/Form.tsx#L50-L83 -> will look into this.